sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/functions/FunctionsTest.java

```diff
@@ -1716,6 +1716,123 @@ public void testIssue1505() {
         );
     }
 
+    @Test
+    public void testArraysOverlap() {
+        this.qs("""
+                SELECT ARRAYS_OVERLAP(ARRAY [1, 2, 3], ARRAY [2, 4]);
+                 arrays_overlap
+                ----------------
+                    true
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(ARRAY [1, 2, 3], cast(null as integer array));
+                 arrays_overlap
+                ----------------
+                    NULL
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(ARRAY [1, 2, 3, null], cast(null as integer array));
+                 arrays_overlap
+                ----------------
+                    NULL
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(cast(null as integer array), ARRAY [1, 2, 3]);
+                 arrays_overlap
+                ----------------
+                    NULL
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(cast(null as integer array), ARRAY [1, 2, 3, null]);
+                 arrays_overlap
+                ----------------
+                    NULL
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(ARRAY [null, 1], ARRAY [2, 1]);
```

Comment:
seems that the description does not match this test.

Reply:
This returns true because there is a common element (1). If there were no common elements, would've returned NULL.

Reply:
Your docs say the because the left array has a null the result is null